spec/acceptance/should_create_task_spec.rb

```diff
@@ -0,0 +1,44 @@
+require 'spec_helper_acceptance'
+
+host = find_only_one("default")
+
+describe "Should create a scheduled task", :node => host do
+
+  before(:all) do
+    @taskname = "pl#{rand(999999).to_i}"
+  end
+
+  after(:all) do
+    on(host, "schtasks.exe /delete /tn #{@taskname} /f", :accept_all_exit_codes => true) do |r|
+      # Empty means deletion was ok.  The 'The system cannot find the file specified' error occurs
+      # if the task does not exist
+      unless r.stderr.empty? || r.stderr =~ /ERROR: The system cannot find the file specified/
+        raise r.stderr
+      end
+    end
+  end
+
+  it "Should create a task if it does not exist", :tier_high => true do
+    pp = <<-MANIFEST
+    scheduled_task {'#{@taskname}':
+      ensure      => present,
+      command     => 'c:\\\\windows\\\\system32\\\\notepad.exe',
+      arguments   => "foo bar baz",
+      working_dir => 'c:\\\\windows',
+      trigger => {
+        schedule   => daily,
+        start_time => '12:00',
+      },
+      provider    => 'taskscheduler_api2'
+    }
+    MANIFEST
+    execute_manifest(pp, :catch_failures => true)
```

Comment:
Can we use `execute_manifest_on` to be more explicit about which nodes are actually executing the manifest?

IMHO, my expectation for that API is that *all* Beaker nodes under test will execute the manifest.  Unfortunately, that's not the case. This may be obvious to people familiar with beaker-testmodeswitcher, but I'd argue it's misleading because it only uses the `default` as @RandomNoun7 and I discovered while reviewing this.  I think this can be a problem when the intent is to execute across a number of hosts (i.e. different Windows versions), and would prefer we remove the implicit `default` for clarity.  Obviously this isn't a problem for masterless when there is only one node, but for all other scenarios (multiple agents, master + agent), etc I think it is.

https://github.com/puppetlabs/beaker-testmode_switcher/blob/c789e6fe7203d19674f5f2181a0d1d8e637cdf11/lib/beaker/testmode_switcher/beaker_runners.rb#L74-L76

Reply:
Updated to `execute_manifest_on`